cleverhans/model_abs.py

```diff
@@ -0,0 +1,138 @@
+from abc import ABCMeta
+
+
+class Model(object):
+    """
+    An abstract interface for model wrappers that exposes model symbols
+    needed for making an attack. This abstraction removes the dependency on
+    any specific neural network package (e.g. Keras) from the core
+    code of CleverHans. It can also simplify exposing the hidden features of a
+    model when a specific package does not directly expose them (needed by
+    "features adversaries").
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        Init a wrapper. If `fprop` is implemented, `__init__`
+        should keep track of the name of the layers or `self.model` should
+        provide a method for retrieving a layer.
+
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
```

Comment:
Should we specify here that this are post-softmax predictions (probabilities) to make it clear what the user should pass? 

To put into context, there is a lot of on-going ambiguity (#160) with the layers that need to be passed so the model abstraction you are contributing is a good opportunity to specify things like that.